src/components/@shared/Price/index.tsx

```diff
@@ -27,7 +27,7 @@ export default function Price({
       conversion={conversion}
       type={accessDetails.type}
     />
-  ) : !accessDetails || accessDetails?.type === '' ? (
+  ) : !accessDetails || !accessDetails?.price || accessDetails?.type === '' ? (
```

Comment:
The logic here needs a rethink. It's already creating a poor user experience where the "No pricing mechanism has been set on this asset yet." message is temporarily shown for all assets until the price has loaded. It's distracting and creates a poor first impression. As described [here](https://github.com/oceanprotocol/market/issues/1378#issuecomment-1114716729)